modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCDataStore.java

```diff
@@ -1855,9 +1856,21 @@ private void insertPS(
                 dialect.onInsert(ps, cx, featureType);
                 ps.addBatch();
             }
-            int[] inserts = ps.executeBatch();
-            checkAllInserted(inserts, features.size());
+            if (keysFetcher.hasPolicy(Policy.pfsequence)) {
+                try {
+                    ps.executeBatch();
+                } catch (AssertionError ae) {
```

Comment:
I don't understand what's going on here, if the batch fails, how are the inserts done? If there is an error, like I don't know, write permissions issues, I want to see an error pop up in the client.